src/Components/App/App.js

```diff
@@ -22,13 +21,25 @@ function App() {
         }
       })
     }, [])
+    
+  const updateSelectedMovie = (event) => {
+    getMovie(event.target.id || event.target.parentNode.id || event.target.parentNode.parentNode.id)
+    .then(data => {
+      if (!data) {
+        setServerError(true)
+      } else {
+        setSelectedMovie(data.movie)
+      }
+    })
+    .then(x => setMovieView(true))
+  }
```

Comment:
good call having setMovieView happen when the state change of selectedMovie resolves